File: Practice 8/Кузнецов Михаил/accessing-data-jpa/src/main/java/com/example/accessingdatajpa/BookController.java
package com.example.accessingdatajpa;

import com.example.accessingdatajpa.model.Book;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping(path = "/books")
public class BookController {
    private BookService bookService = new BookService();

    @GetMapping(value = "{isbn}")
    public @ResponseBody
    Book getBook(@PathVariable Long isbn) {
        return bookService.read(isbn);
    }

    @PostMapping
    public void postBook(@RequestBody Book book) {
        bookService.create(book);
    }

    @PatchMapping
    public @ResponseBody
    boolean patchBook(@RequestBody Book book) {
        return bookService.update(book);
    }

    @DeleteMapping(value = "{isbn}")
    public @ResponseBody
    boolean deleteBook(@PathVariable Long isbn) {
        return bookService.delete(isbn);
    }

}

File: Practice 8/PopovOO/libraryBD/src/main/kotlin/ru/hse/librarybd/contoller/BookController.kt
package ru.hse.librarybd.contoller

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.hse.librarybd.contoller.abstraction.EntityController
import ru.hse.librarybd.dto.Book
import ru.hse.librarybd.dto.BookDto
import ru.hse.librarybd.service.BookService

@RestController
@RequestMapping("book")
class BookController(
    bookService: BookService
) : EntityController<Book, BookDto, String>(bookService)
